examples/playground/overlays/example-append-map-to-array/append-map-to-array.yml

```diff
@@ -0,0 +1,15 @@
+#@ load("@ytt:overlay", "overlay")
+
+#! Add a map to an array of maps:
+
+#@overlay/match by=overlay.all
+---
+awards:
+  books:
+    #@overlay/match by=overlay.all, expects="1+"
+    -
+      #@overlay/match missing_ok=True
+      reviewers:
+        #@overlay/append
+        - id: 1
```

Comment:
When this example runs, it outputs the following as the first array item under the `books` key:
```
- id: 1
    title: International Botev
    reviewers:
    - id: 2
      name: PersonB
    - id: 1
      name: Person A
```

Because we're `@overlay/append`ing it puts the second id at the end of the `reviewers` array (after the first id). This may be confusing for people who see the ids out of order. Instead, could we replace the `id` field with something non-order dependent?